boa/interpret.py

```diff
@@ -86,22 +91,25 @@ def _ifaces():
 
     if _disk_cache is None:
         ifaces = _ifaces()
-        ret = CompilerData(source_code, contract_name, interface_codes=ifaces, **kwargs)
-        return ret
+        return deployer.create_compiler_data(
+            source_code, contract_name, interface_codes=ifaces, **kwargs
+        )
 
     def func():
         ifaces = _ifaces()
-        ret = CompilerData(source_code, contract_name, interface_codes=ifaces, **kwargs)
+        ret = deployer.create_compiler_data(
+            source_code, contract_name, interface_codes=ifaces, **kwargs
+        )
         with anchor_compiler_settings(ret):
             _ = ret.bytecode, ret.bytecode_runtime  # force compilation to happen
         return ret
 
-    cache_key = str((contract_name, source_code, kwargs))
+    cache_key = str((contract_name, source_code, kwargs, deployer))
```

Comment:
do we know that `deployer` has a sane repr? it could be any callable. we should at least have it on a separate line to be a bit clearer
```suggestion
    assert isinstance(deployer, type)
    deployer_id = repr(deployer)  # arbitrarily choose a unique str for the deployer
    cache_key = str((contract_name, source_code, kwargs, deployer_id))
```

Reply:
if it's a function, the repr will have a pointer. it is more likely to produce a false cache miss than a false cache hit, but it's still possible.